sqlmesh/core/context.py

```diff
@@ -1644,13 +1654,22 @@ def plan_builder(
         elif forward_only is None:
             forward_only = self.config.plan.forward_only
 
+        # When handling prod restatements, only clear intervals from other model versions if we are using full virtual environments
+        # If we are not, then there is no point, because none of the data in dev environments can be promoted by definition
+        clear_restated_intervals_across_model_versions = (
```

Comment:
I'd call this `restate_shared_version_snapshots` or `restate_snapshots_with_same_version`